x-pack/plugins/apm/public/components/app/service_groups/service_group_save/select_services.tsx

```diff
@@ -78,6 +76,19 @@ export function SelectServices({
     }
   }, [isEdit, serviceGroup.kuery]);
 
+  useEffect(() => {
+    if (!stagedKuery) {
+      return;
+    }
+    const { isValid, isParsingError, message } =
+      validateServiceGroupKuery(stagedKuery);
+    if (isValid || isParsingError) {
```

Comment:
If we don't care about the difference between `isValid` and `isParsingError` why have both? Can we simplify and just have `isValid` (YAGNI)?

Reply:
The problem with combining them is `validateServiceGroupKuery` is used in both the route and the client-side routing. In the route, we only care if the KQL is valid (i.e. is is parsed and only contains valid fields). However, in the client, incomplete KQL (parsing errors) should be ignored because the user is in the middle of typing out a full kuery filter despite the fact that `isValid` is not true (because it can't be parsed).

Reply:
Okay, that makes sense. But we still don't care about the difference here. We just want to update the message